### PR TITLE
[BUILD] Working Docker image and using it for Travis CI

### PR DESCRIPTION
**Pull request by AVSurfer123**
We now have a working docker image from https://hub.docker.com/repository/docker/avsurfer123/c-runtime. The Travis CI uses this image to check if the code is working. The test currently run is just running `tcp_client` and `udp_client` once. But eventually we will replace this with the new `net_handler` tests.

### CODE REVIEW

**Comment by benliao1** [comment]
Also, are we going to try to have that global Makefile, like in the issue that you linked? I almost feel like that's not really necessary any more, if we're going to be building stuff in the Docker. Or we can have a script do it for us, like:
`cd shm_wrapper && make` and so on...

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    popd && rm -r ${protobuf_c_folder}
+
+RUN apt-get autoremove && apt-get clean
+
+# Debug
+RUN apt-get -y install procps htop tmux nano
+
+WORKDIR /root/runtime
+
+ADD ./ ./
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply Dockerfile:46]: Adds all the files from the CWD, which is `c-runtime/` that are not in `.dockerignore`.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# branches:
+#   only:
+#     - master
+
+if: (type = push AND branch = master) OR (type = pull_request)
+
+before_install:
+  - docker pull ${DOCKER_USERNAME}/c-runtime:arm32
+
+script:
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply .travis.yml:20]: So you don't actually need to expose those ports for CI so I'll remove them. The docker ps just gets the container ID

File: executor/Makefile [hunk 3]
```diff
@@ -8,7 +8,7 @@ PY_LIB=-l$(py)m
 
 UNAME=$(shell uname)
 ifeq ($(UNAME), Linux)
-LIBS=-pthread -lrt -export-dynamic
+LIBS=-pthread -lrt -export-dynamic -fPIC
 #-rdynamic or -export-dynamic to make it dynamically linked
```
[the remark this answers is not in the document]
Comment (AVSurfer123) [reply executor/Makefile:12]: I actually don't know what it does but executor fails to compile in the docker container without it. Says some error about shared objects.

**Comment by AVSurfer123** [comment]
I just added a script in the root directory `./build.sh` which will build everything so we don't need a global Makefile anymore.

File: executor/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (benliao1) [reply executor/Makefile:12]: Hm, ok, seems like from the documentation it says generate position-independent code (PIC), not sure what it means..

For future reference: https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#Code-Gen-Options (search `-fPIC`)

[the remark this answers is not in the document]
Comment (AVSurfer123) [reply executor/Makefile:12]: This is the error that I get without the flag 
```
/usr/bin/ld: /tmp/ccROSw9C.o: relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /tmp/ccCNVRJN.o: relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /tmp/ccZFdNzq.o: relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /tmp/ccsD6CXc.o: relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /tmp/ccROSw9C.o(.text+0x10): unresolvable R_ARM_THM_CALL relocation against symbol `pthread_self@@GLIBC_2.4'
/usr/bin/ld: final link failed: nonrepresentable section on output
collect2: error: ld returned 1 exit status
```

**Pull request merged**